flow/layers/container_layer.h

```diff
@@ -114,11 +115,14 @@ class MergedContainerLayer : public ContainerLayer {
  public:
   MergedContainerLayer();
 
-  void AssignOldLayer(Layer* old_layer) override;
-
   void Add(std::shared_ptr<Layer> layer) override;
 
  protected:
+#ifdef FLUTTER_ENABLE_DIFF_CONTEXT
+  void DiffChildren(DiffContext* context,
```

Comment:
The base method is public, should the override here be public too?

Reply:
According to @chinmaygarde he actually prefers what you did here to my suggestion...

Reply:
Right, though I should clarify that both ways will work. I like this approach because the subclass decides what the visibility of its further subclasses ought to be. Unless it is final. In that case, it doesn't matter either way.

You won't be able to do this is Java though as the compiler would be mad. This pattern violates Liskov substitution.

Reply:
I'm going to go with @flar here. I didn't mean for the override to have different visibility than the original method. I don't have any strong feelings either way in this case, but I also don't see any reason right now to limit visibility for potential subclasses.